handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -692,7 +692,7 @@ public void handlerRemoved0(ChannelHandlerContext ctx) throws Exception {
             }
             if (!sslClosePromise.isDone()) {
                 if (cause == null) {
-                    cause = new SSLHandshakeException("SslHandler removed before handshake completed");
+                    cause = new SSLHandshakeException("SslHandler removed before SSLEngine was closed");
```

Comment:
I wonder whether we need to use something other than `SSLHandshakeException` taking to account it's not checked and probably not related .